implementation/src/main/java/io/smallrye/config/mapper/ConfigMappingProvider.java

```diff
@@ -0,0 +1,703 @@
+package io.smallrye.config.mapper;
+
+import static io.smallrye.config.mapper.ConfigurationInterface.GroupProperty;
+import static io.smallrye.config.mapper.ConfigurationInterface.LeafProperty;
+import static io.smallrye.config.mapper.ConfigurationInterface.MapProperty;
+import static io.smallrye.config.mapper.ConfigurationInterface.MayBeOptionalProperty;
+import static io.smallrye.config.mapper.ConfigurationInterface.PrimitiveProperty;
+import static io.smallrye.config.mapper.ConfigurationInterface.Property;
+import static io.smallrye.config.mapper.ConfigurationInterface.getConfigurationInterface;
+
+import java.io.Serializable;
+import java.lang.reflect.Method;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+
+import org.eclipse.microprofile.config.spi.Converter;
+
+import io.smallrye.common.constraint.Assert;
+import io.smallrye.common.function.Functions;
+import io.smallrye.config.ConfigValidationException;
+import io.smallrye.config.KeyMap;
+import io.smallrye.config.NameIterator;
+import io.smallrye.config.SmallRyeConfig;
+
+/**
+ *
+ */
+public final class ConfigMappingProvider implements Serializable {
```

Comment:
Given that the config mapping is now so strongly integrated within the config, I'm not sure it makes sense to have this still be public - it really looks like it's just an implementation detail of `SmallRyeConfigBuilder` at this point.

Reply:
We need to get rid of the `mapper` package to change the modifier.